From 1bf46e951b9d35e2e5ad61c85b15f32496017247 Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Tue, 7 Feb 2017 16:02:29 +0100
Subject: [PATCH] Use StateChanged signal instead of StateChange

StateChange signal has been deprecated since NM 0.7 and has been
completely removed in 0.9, see:
https://developer.gnome.org/NetworkManager/0.9/ref-migrating.html

Closes: #58
---
 libproxy/modules/network_networkmanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libproxy/modules/network_networkmanager.cpp b/libproxy/modules/network_networkmanager.cpp
index 5198860..546a8ee 100644
--- a/libproxy/modules/network_networkmanager.cpp
+++ b/libproxy/modules/network_networkmanager.cpp
@@ -62,7 +62,7 @@ class networkmanager_network_extension : public network_extension {
 
 			// If connection was successful, set it up
 			dbus_connection_set_exit_on_disconnect(conn, false);
-			dbus_bus_add_match(conn, "type='signal',interface='" NM_DBUS_INTERFACE "',member='StateChange'", NULL);
+			dbus_bus_add_match(conn, "type='signal',interface='" NM_DBUS_INTERFACE "',member='StateChanged'", NULL);
 			dbus_connection_flush(conn);
 		}
 
From 0bdcf9f9dbee67dee47694e0012fa891588abc89 Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Tue, 7 Feb 2017 16:05:00 +0100
Subject: [PATCH] Set lasturl to NULL after free/delete

It avoids the lasturl to be freed a 2nd time when the rewind() method is
called.

Closes: #59
---
 libproxy/modules/wpad_dns_alias.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libproxy/modules/wpad_dns_alias.cpp b/libproxy/modules/wpad_dns_alias.cpp
index 669dccf..5b8c2e0 100644
--- a/libproxy/modules/wpad_dns_alias.cpp
+++ b/libproxy/modules/wpad_dns_alias.cpp
@@ -39,6 +39,7 @@ class dns_alias_wpad_extension : public wpad_extension {
 		lastpac = *pac = lasturl->get_pac();
 		if (!lastpac) {
 		    delete lasturl;
+		    lasturl = NULL;
 		    return NULL;
 		}
 
